DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs

```diff
@@ -0,0 +1,127 @@
+            if (versions.Count == 1)
+            {
+                _notifier.Warning(_h["There is currently only one version of this content item."]);
+            }
+
+            var contentItem = versions.First().ContentItem;
+
+            var selectVersions = BuildSelectList(versions.Select(v => v.VersionNumber).ToList());
+
+            var options = new VersionComparisonOptions
+            {
+                ContentItemId = contentItemId,
+                BaseVersion = selectVersions[0].Value,
+                BaseVersionSelectListItems = selectVersions,
+                CompareVersion = versions.Count > 1 ? selectVersions[1].Value : selectVersions[0].Value,
+                CompareVersionSelectListItems = selectVersions
+            };
+
+            var shapeViewModel = await GetShapeViewModel(options, contentItem);
+
+            return View(shapeViewModel);
+        }
+
+        [HttpPost, ActionName("Index")]
+        public async Task<ActionResult> IndexPost([Bind(Prefix = "VersionComparisonOptions")] VersionComparisonOptions options)
+        {
+            if (options.BaseVersion == options.CompareVersion)
+            {
+                _notifier.Warning(_h["Both the base and compare version are the same."]);
+            }
+
+            var versions = await _auditTrailQueryService.GetVersions(options.ContentItemId ?? string.Empty);
+            var contentItem = versions.First().ContentItem;
+            var selectVersions = BuildSelectList(versions.Select(v => v.VersionNumber).ToList());
+
+            options.BaseVersionSelectListItems = selectVersions;
+            options.CompareVersionSelectListItems = selectVersions;
+
+            var shapeViewModel = await GetShapeViewModel(options, contentItem);
+
+            return View(shapeViewModel);
+        }
+
+        private async Task<IShape> GetShapeViewModel(VersionComparisonOptions options, ContentItem contentItem)
+        {
+            var selectListShape =
+                await _versionComparisonOptionsDisplayManager.BuildEditorAsync(options,
+                    _updateModelAccessor.ModelUpdater, false);
+
+
+            return await _shapeFactory.CreateAsync<VersionComparisonViewModel>("VersionComparison",
+                viewModel =>
+                {
+                    viewModel.ContentItemId = options.ContentItemId;
+                    viewModel.ContentItemDisplayName = contentItem.DisplayText;
+                    viewModel.ContentItemContentType = contentItem.ContentType;
+                    viewModel.SelectLists = selectListShape;
+                });
+        }
+
+        private List<SelectListItem> BuildSelectList(List<int> versionNumbers)
+        {
+            var selectListItems = new List<SelectListItem>();
+            for (int i = 0; i < versionNumbers.Count; i++)
+            {
+                var versionNumberValue = versionNumbers[i].ToString();
+                var versionNumberText = string.Format("version {0}{1}", versionNumberValue,
```

Comment:
string interpolation?

Reply:
A bit subjective which one is more readable with the ternery operation but have changed to string interpolation.